source/config.go

```diff
@@ -35,6 +41,23 @@ type Config struct {
 }
 
 // Validate executes manual validations beyond what is defined in struct tags.
-func (c Config) Validate() error {
-	return c.Config.Validate()
+func (c *Config) Validate(ctx context.Context) error {
+	err := c.Config.Validate()
+	if err != nil {
+		return err
+	}
```

Comment:
We should combine the errors coming from `c.Config.Validate` with the topic validation (see how we do it in `common.Config.Validate`).